src/external/etsf_nanoquanta.jl

```diff
@@ -146,18 +154,29 @@ load_basis(folder; kwargs...) = load_basis(Float64, folder; kwargs...)
 Load a DFTK-compatible density object from the ETSF folder.
 Use the scalar type `T` to represent the data.
 """
-function load_density(T, folder::EtsfFolder)
-    # So far this function is untested
-    basis = load_basis(T, folder)
+function load_density(T, folder::EtsfFolder; magnetic_moments=[])
+    basis = load_basis(T, folder, magnetic_moments=magnetic_moments)
 
-    n_comp = size(folder.den["density"], 5)
     n_real_cpx = size(folder.den["density"], 1)
-    @assert n_comp == 1
     @assert n_real_cpx == 1
 
-    ρ_real = Array{Complex{T}}(folder.den["density"][1, :, :, :, 1])
-    @assert basis.fft_size == size(ρ_real)
+    n_spin = size(folder.den["density"], 5)
+    @assert n_spin in (1, 2)
 
-    from_real(basis, ρ_real)
+    if n_spin == 1
+        @assert basis.model.spin_polarization == :none
+        ρtot_real = Array{T}(folder.den["density"][1, :, :, :, 1])
+        @assert basis.fft_size == size(ρ_real)
+
+        return from_real(basis, ρtot_real), nothing
+    else
+        @assert basis.model.spin_polarization == :collinear
+        ρtot_real = Array{T}(folder.den["density"][1, :, :, :, 1])
+        ρα_real = Array{T}(folder.den["density"][1, :, :, :, 2])
```

Comment:
They really have rhotot and rhoalpha ?!?! Crazy people.

Reply:
Yes ... that surprised me as well.